### PR TITLE
[xmlsec] bump deps + build requires

### PR DESCRIPTION
**Pull request by dmn-star**
Specify library name and version:  **xmlsec/all**

This is also a good place to share with all of us **why you are submitting this PR** (specially if it is a new addition to ConanCenter): is it a dependency of other libraries you want to package? Are you the author of the library? Thanks!

---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Review by uilianries — APPROVED** [review]

**Review by prince-chrismc — APPROVED** [review]

File: recipes/xmlsec/all/conanfile.py [hunk 1]
```diff
@@ -54,9 +54,9 @@ def configure(self):
 
     def build_requirements(self):
         if not self._is_msvc and tools.os_info.is_windows and not tools.get_env("CONAN_BASH_PATH"):
```
Comment (prince-chrismc) [inline recipes/xmlsec/all/conanfile.py:56]: I think there's an update to this line 🤔 @madebr or @SpaceIm if you have a second to check

**Review by SSE4 — APPROVED** [review]

File: recipes/xmlsec/all/conanfile.py (hunk 1 above)
Comment (SpaceIm) [reply recipes/xmlsec/all/conanfile.py:56]: yes:

```python
    @property
    def _settings_build(self):
        getattr(self, "settings_build", self.settings)

    def build_requirements(self):
        if not self._is_msvc and self._settings_build.os == "Windows" and not tools.get_env("CONAN_BASH_PATH"):
            self.build_requires("msys2/cci.latest")
```

File: recipes/xmlsec/all/conanfile.py [hunk 2]
```diff
@@ -54,9 +54,9 @@ def configure(self):
...
-            self.build_requires("msys2/20200517")
+            self.build_requires("msys2/cci.latest")
         self.build_requires("libtool/2.4.6")
-        self.build_requires("pkgconf/1.7.3")
```
Comment (SpaceIm) [inline recipes/xmlsec/all/conanfile.py:59]: Does msvc need libtool and pkgconf?

File: recipes/xmlsec/all/conanfile.py (hunk 1 above)
Comment (SpaceIm) [reply recipes/xmlsec/all/conanfile.py:56]: https://github.com/conan-io/conan-center-index/pull/6853

**Pull request merged**